packages/product-json-to-csv/test/main.spec.js

```diff
@@ -0,0 +1,625 @@
+import StreamTest from 'streamtest'
+import { oneLineTrim } from 'common-tags'
+import ProductJsonToCsv from '../src/main'
+import * as writer from '../src/writer'
+
+const streamTest = StreamTest.v2
+
+describe('ProductJsonToCsv', () => {
+  let productJsonToCsv
+  beforeEach(() => {
+    const logger = {
+      error: () => {},
+      warn: () => {},
+      info: () => {},
+      verbose: () => {},
+    }
+    const parserConfig = {
+      categoryBy: 'name',
+      categoryOrderHintBy: 'name',
+      delimiter: ',',
+      fillAllRows: false,
+      language: 'en',
+      multiValueDelimiter: ';',
+      productSeparator: '\n',
+    }
+    productJsonToCsv = new ProductJsonToCsv(
+      { projectKey: 'project-key' },
+      parserConfig,
+      logger,
+      'myAccessToken'
+    )
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  describe('::constructor', () => {
+    it('should initialize with defaults', () => {
+      const apiConfig = {
+        projectKey: 'foo',
+      }
+      const defaultConfig = {
+        delimiter: ',',
+        multiValueDelimiter: ';',
+        fillAllRows: false,
+        categoryBy: 'name',
+        categoryOrderHintBy: 'name',
+        language: 'en',
+        productSeparator: '\n',
+      }
+
+      productJsonToCsv = new ProductJsonToCsv(apiConfig)
+      expect(productJsonToCsv.logger).toBeDefined()
+      expect(productJsonToCsv.client).toBeDefined()
+      expect(productJsonToCsv.parserConfig).toEqual(defaultConfig)
+    })
+  })
+
+  describe('::run', () => {
+    let zipSpy
+    let csvSpy
+    beforeEach(() => {
+      zipSpy = jest.spyOn(writer, 'writeToZipFile').mockImplementation(() => {})
```

Comment:
Automocking applies here to and eases things. This should have an example. https://github.com/tdeekens/flopflip/blob/master/packages/splitio-adapter/modules/adapter/adapter.spec.js